### PR TITLE
feat: add more features and solve some bugs on pagination component

### PR DESCRIPTION
**Pull request by maik-emanoel**
Some features I did:

Make the select work;
Solve bugs when the user was gonna to the last page and changed the select value;
Let all the info dynamic;

It's my first time make I PR, so I'm sorry some mistake.

### CODE REVIEW

File: src/app.tsx [hunk 1]
```diff
@@ -156,7 +159,13 @@ export function App() {
...
         </Table>
 
-        {tagsResponse && <Pagination pages={tagsResponse.pages} items={tagsResponse.items} page={page} />}
+        {tagsResponse && (
+          <Pagination 
+            pages={tagsResponse.pages} 
+            items={tagsResponse.items} 
+            page={page} 
+            itemsPerPage={tagsResponse.data}/>
+          )}
```
Comment (filipedev6) [inline src/app.tsx]:
```suggestion
          <Pagination 
            pages={tagsResponse.pages} 
            items={tagsResponse.items} 
            page={page} 
            itemsPerPage={per_page}/>
          )}
```

File: src/components/pagination.tsx [hunk 2]
```diff
@@ -8,15 +8,33 @@ import {
...
 import { Button } from './ui/button'
 import { Select, SelectContent, SelectItem, SelectTrigger } from './ui/select'
+import { Tag } from '../app'
+import { useEffect } from 'react'
 
 interface PaginationProps {
   pages: number
   items: number
   page: number
+  itemsPerPage: Tag[]
```
Comment (filipedev6) [inline src/components/pagination.tsx]:
```suggestion
  itemsPerPage: number
```

File: src/components/pagination.tsx [hunk 3]
```diff
@@ -8,15 +8,33 @@ import {
...
 }
 
-export function Pagination({ items, page, pages }: PaginationProps) {
-  const [, setSearchParams] = useSearchParams()
+export function Pagination({ items, page, pages, itemsPerPage }: PaginationProps) {
+  const [searchParams, setSearchParams] = useSearchParams()
+
+  const rowsPerPage = searchParams.get("per_page")
+  ? Number(searchParams.get("per_page"))
+  : 10;
```
Comment (filipedev6) [inline src/components/pagination.tsx]:
```suggestion
```

There is no need to create code that has already been created in the parent component.

File: src/components/pagination.tsx [hunk 4]
```diff
@@ -58,14 +76,24 @@ export function Pagination({ items, page, pages }: PaginationProps) {
...
+      params.set("per_page", rows);
+
+      return params;
+    });
+  }
+
   return (
     <div className="flex text-sm items-center justify-between text-zinc-500">
-      <span>Showing 10 of {items} items</span>
+      <span>Showing {itemsPerPage.length} of {items} items</span>
```
Comment (filipedev6) [inline src/components/pagination.tsx]:
```suggestion
      <span>Showing {itemsPerPage} of {items} items</span>
```

File: src/components/pagination.tsx [hunk 5]
```diff
@@ -58,14 +76,24 @@ export function Pagination({ items, page, pages }: PaginationProps) {
...
       <div className="flex items-center gap-8">
         <div className="flex items-center gap-2">
           <span>Rows per page</span>
 
-          <Select defaultValue="10">
+          <Select defaultValue={String(rowsPerPage)} onValueChange={handleNumberOfRows}>
```
Comment (filipedev6) [inline src/components/pagination.tsx]:
```suggestion
          <Select defaultValue={String(itemsPerPage)} onValueChange={handleNumberOfRows}>
```

**Review by filipedev6 — CHANGES_REQUESTED** [review]

**Comment by filipedev6** [comment]
Contribute some changes to your project, don't get me wrong but there are some important points that make a difference. Nice to meet you, I hope you are happy with my contributions!

**Review by maik-emanoel — COMMENTED** [review]
Good changes

**Comment by filipedev6** [comment]
@diego3g approve this PR 🚀

**Pull request closed without merging**